plugin-server/src/ingestion/ingestion-consumer.test.ts

```diff
@@ -0,0 +1,265 @@
+        // TRICKY: This is the slightly different format that capture sends
+        const captureEvent = {
+            uuid: event.uuid,
+            distinct_id: event.distinct_id,
+            ip: event.ip,
+            now: event.now,
+            token: event.token,
+            data: JSON.stringify(event),
+        }
+        return {
+            value: Buffer.from(JSON.stringify(captureEvent)),
+            size: 1,
+            topic: 'test',
+            offset: offsetIncrementer++,
+            partition: 1,
+        }
+    })
+}
+
+describe('IngestionConsumer', () => {
+    let ingester: IngestionConsumer
+    let hub: Hub
+    let team: Team
+    let team2: Team
+
+    const createEvent = (event?: Partial<PipelineEvent>): PipelineEvent => ({
+        distinct_id: 'user-1',
+        uuid: new UUIDT().toString(),
+        token: team.api_token,
+        ip: '127.0.0.1',
+        site_url: 'us.posthog.com',
+        now: DateTime.now().toISO(),
+        event: '$pageview',
+        properties: {
+            $current_url: 'http://localhost:8000',
+        },
+        ...event,
+    })
+
+    beforeEach(async () => {
+        const now = new Date(2025, 1, 1).getTime()
+        jest.spyOn(Date, 'now').mockReturnValue(now)
+        offsetIncrementer = 0
+        await resetTestDatabase()
+        hub = await createHub()
+
+        hub.kafkaProducer = mockProducer
+        team = await getFirstTeam(hub)
+        const team2Id = await createTeam(hub.db.postgres, team.organization_id)
+        team2 = (await hub.db.fetchTeam(team2Id)) as Team
+    })
+
+    afterEach(async () => {
+        jest.restoreAllMocks()
+        jest.setTimeout(10000)
+        if (ingester) {
+            await ingester.stop()
+        }
+        await closeHub(hub)
+    })
+
+    afterAll(() => {
+        jest.useRealTimers()
+    })
+
+    describe('general', () => {
+        beforeEach(async () => {
+            ingester = new IngestionConsumer(hub)
+            await ingester.start()
+        })
+
+        it('should have the correct config', () => {
+            expect(ingester['name']).toMatchInlineSnapshot(`"ingestion-consumer-events_plugin_ingestion_test"`)
+            expect(ingester['groupId']).toMatchInlineSnapshot(`"events-ingestion-consumer"`)
+            expect(ingester['topic']).toMatchInlineSnapshot(`"events_plugin_ingestion_test"`)
+            expect(ingester['dlqTopic']).toMatchInlineSnapshot(`"events_plugin_ingestion_dlq_test"`)
+            expect(ingester['overflowTopic']).toMatchInlineSnapshot(`"events_plugin_ingestion_overflow_test"`)
+        })
+
+        it('should process a standard event', async () => {
+            await ingester.handleKafkaBatch(createKafkaMessages([createEvent()]))
+
+            expect(forSnapshot(getProducedKafkaMessages())).toMatchSnapshot()
+        })
+    })
+
+    describe('dropping events', () => {
+        describe.each(['headers', 'payload'] as const)('via %s', (kind) => {
```

Comment:
note: For me it's quite a bit of effort to understand those tests – there's quite a lot of back-and-forth:

- to understand why the payload tests pass, I had to scroll up quite a bit to find that there's a default token in the payload
- in the header tests the payloads still have the tokens – the name of the describe block suggest that we're testing events with tokens only in the headers

The tests below are short enough (and very readable on their own) that they could be duplicated and the `createEvent` function could be explicit on where the token lands, instead of relying on `addMessageHeaders` filtering by the kind of test.

Reply:
Cool point taken. I think its too early for this to be reviewed tbh as I'm mostly trying to understand the existing logic myself...